packages/dds/map/src/directory.ts

```diff
@@ -1254,10 +1254,14 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 		if (key === undefined || key === null) {
 			throw new Error("Undefined and null keys are not supported");
 		}
+
 		const previousOptimisticLocalValue = this.getOptimisticValue(key);
 
-		// Create a local value and serialize it.
-		bindHandles(value, this.serializer, this.directory.handle);
+		if (this.runtime.options.allowDetachedResolve) {
```

Comment:
you should use config, not options. config is available off a monitor context, usually a member called mc. you can create an monitoring context off a logger if there isn't already one. there are lots of examples in the code.